### PR TITLE
feat: add `verifyBody` to customize the mismatch rules for response body

### PR DESCRIPTION
**Pull request by GuanyuChen**
Hey bro. 
This PR is about customize the mismatch rules for response body.  And i added samples and unit tests. Please contact me if you have any questions. 
tks so much

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/httpClient.js [hunk 1]
```diff
@@ -91,15 +91,18 @@ function Client (opts) {
...
-      }
-
       this.requestIterator.recordBody(resp.req, resp.headers.statusCode, resp.body, resp.headers.headers)
 
       this.emit('response', resp.headers.statusCode, resp.bytes, resp.duration, this.rate)
 
       this._doRequest()
+
+      const isFn = typeof this.opts.verifyBody === 'function'
+      if (isFn && !this.opts.verifyBody(resp.body)) {
```
Comment (GuanyuChen) [inline lib/httpClient.js:101]: I think executing the verify function later can improve performance

**Comment by GuanyuChen** [comment]
It's my honor. Looking forward to the code being merged soon

**Comment by GuanyuChen** [comment]
> lgtm

**Comment by GuanyuChen** [comment]
I'm very sorry to bother you. Are there any other issues that need improvement? My project needs this feature very much 🙏🏻

**Comment by mcollina** [comment]
I'm currently on vacation for the next two weeks. I'll try to ship but can't guarantee.

**Comment by GuanyuChen** [comment]
> I'm currently on vacation for the next two weeks. I'll try to ship but can't guarantee.

get it. I'll patch it temporarily. have a good holiday!

**Pull request merged**